Added  Movie Creator Tool

prathamjoshi opened the pull request:
Added file `mpegCreator.py' that creates movies from the wide and center images using`ffmpeg` tool. Can specify particular folder or have it run for all.

prathamjoshi committed:
added mpegCreator.py, initial version

prathamjoshi committed:
updated script. working but with various ffmpeg issues

prathamjoshi committed:
updated script. working but with various ffmpeg issues

prathamjoshi committed:
now mpegCreation script should work properly

prathamjoshi committed:
updating script before refactor

prathamjoshi committed:
Refactored mpegCreator.py

acviana commented on tools/mpegCreator.py:
```diff
@@ -0,0 +1,105 @@
+import os 
+
+
+PATH = "/astro/3/mutchler/mt/drizzled" #image path
```
`PATH` here is a global variable because it is not defined inside any function, class, or loop. All-Caps is the correct convention for this.

acviana commented on tools/mpegCreator.py:
```diff
@@ -0,0 +1,105 @@
+
+PATH = "/astro/3/mutchler/mt/drizzled" #image path
+
+global source
```
Here `source` is already a global variable so `global` doesn't do anything. You use `global` if you want to make a local variable (say one defined inside a function) global. But declaring global variables that way can be a pain when it comes to debugging. Finally, `source` should be in all caps by convention because it's a global variable. More reading: http://stackoverflow.com/questions/4693120/use-of-global-keyword-in-python

acviana commented on tools/mpegCreator.py:
```diff
@@ -0,0 +1,105 @@
+if __name__ == '__main__':
+    args = parse_args() 
+    source = args.source 
+    currentDir = os.getcwd() 
```
`currentDir` is never used again. I think this can be deleted (pylint would have caught this).

acviana commented on tools/mpegCreator.py:
```diff
@@ -0,0 +1,105 @@
+    all the subfolders and calls the runScript function accordingly.
+    If no subfolder given calls runScript iteratively on all subfolders.
+    """
+    path = PATH 
```
The global variable `PATH` does not need to be reassigned to the local variable `path` because variables in the global scope are global.

prathamjoshi replied:
I am editing local `path` with respect to global `PATH` . have local `path` because that changes but the global `PATH` does not (path to the main directory with all the image folders)

acviana commented on tools/mpegCreator.py:
```diff
@@ -0,0 +1,105 @@
+    buildMovies(movieType, path)
+    movieType = 'center'
+    buildMovies(movieType, path)
+    
```
So here you are defining variables that you never use again. You could consolidate and clarify this by:

``` python
buildMovies( 'wide', path)
buildMovies('center', path)
```

acviana commented on tools/mpegCreator.py:
```diff
@@ -0,0 +1,105 @@
+    try:
+        os.chdir(path)
+    except:
+        return 
```
Why do we have this try/except statement here?

prathamjoshi replied:
This is to prevent the program from crashing if it cannot find/access a particular png directory while operating on multiple sources. For example, on our drizzled folder I did not have access to the `vesta` png folders and the program simply ignores it and continues instead of crashing. There might be better ways to do this but this was the first thing that came to  mind. I am open to other suggestions.

acviana commented:
Hi Pratham. Good work, the script is coming along. I have some additional comments, please feel free to discuss them if you disagree with anything.

prathamjoshi commented:
Thanks for the comments. Will go over them tomorrow morning and reply.

prathamjoshi committed:
fixed issues with mpegCreator with respect to the comments

prathamjoshi commented:
So Alex, I think I have fixed most of the issues highlighted above. I have fixed the issues with the global variables (thanks for the links I feel I understand variable scopes in `python` better now). I am leaving the local `path` and the try/except statements for reasons highlighted above. 
Also, ran through `pylint` the score is still only 5.74 but it is mostly things with line-length and variable naming conventions that I can fix easily once you approve on the code.
Let me know what you think.

prathamjoshi committed:
made the suggested changes in mpegCreator.py and tested the script out
